logging/src/main/java/alluxio/logging/base/FlowLoggingAspect.java

```diff
@@ -0,0 +1,88 @@
+package alluxio.logging.base;
+
+import br.com.simbiose.debug_log.BaseAspect;
+import org.aspectj.lang.ProceedingJoinPoint;
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * A base class for all flow logging aspects. This class defines basic
+ * logging aspect operations needed for {@link BaseAspect} to successfully
+ * log the desired flows.
+ */
+public abstract class FlowLoggingAspect extends BaseAspect {
+
+    // A fake thread ID needed for a flow not to be interrupted
+    // in between threads with different IDs
+    protected static final long artificialThreadId = -1L;
+
+    // Keeps the ID of the thread currently executing a method to be logged
+    protected final Map<Long, Integer> threadIdToStep = new ConcurrentHashMap<>();
+
+    // Maps thread ID to flow ID
+    protected final Map<Long, Long> threadIdToDebugLogId = new ConcurrentHashMap<>();
+
+    /**
+     * Starts logging all methods within the white/blacklist once the initial method is called.
+     *
+     * @param point the {@link ProceedingJoinPoint} for the {@code proceed(..)} method
+     *        in AspectJ
+     * @return the same object of the method being wrapped by AspectJ
+     * @throws Throwable the same {@code Throwable} thrown bt the method
+     *         being wrapped by AspectJ
+     */
+    public Object startFlux(final ProceedingJoinPoint point) throws Throwable {
```

Comment:
Please, let's never use the word "flux", always use "flow".
Please, do it for all methods you created with this name.